Person.rb

```diff
@@ -0,0 +1,32 @@
+class Person
+  attr_reader :id, :name, :age
+
+  def initialize(name = "Unknown", age = 0, parent_permission = true)
+    @id = generate_id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def name=(name)
+    @name = name
+  end
+
+  def age=(age)
+    @age = age
+  end
```

Comment:
- Optional.
I would recommend removing these setters as you can use the `attr_accessor` to achieve the same result.